### PR TITLE
Fix #1147

### LINKED ISSUE #1147: [Bug] Lua wonkiness
The MoonSharp interpreter vomits out errors:

Stack trace:
ScriptRuntimeException: cannot access field powerValue of userdata<Furniture>
MoonSharp.Interpreter.Execution.VM.Processor.ExecIndex (MoonSharp.Interpreter.Execution.VM.Instruction i, Int32 instructionPtr)
MoonSharp.Interpreter.Execution.VM.Processor.Processing_Loop (Int32 instructionPtr)

Steps to recreate: 
-Hover over any piece of furniture

### PR DESCRIPTION
**Pull request by Grenkin1988**
Fixed problem related to PowerSystem

There is still one issue:
image

### CODE REVIEW

**Comment by mikejbrown** [comment]
The new bug seems to be that `powerValueAt` is returning values outside the valid range of 0 <= v < 256. Put:
```
private void GenerateTexture()
{
///... stuff ...
            if( (v >= 0 && v < 256) == false)
            {
                Debug.ULogChannel("OverlayMap", "v is out of bounds value '{0}'", v);
                v = 0;
            }
            Debug.Assert(v >= 0 && v < 256, "v >= 0 && v < 256");
```
in OverlayMap.cs to confirm this is the issue. So you need to clamp the output or rescale it or something.

**Comment by Grenkin1988** [comment]
Fixed texture issue

File: Assets/StreamingAssets/Overlay/overlay_functions.lua [hunk 1]
```diff
@@ -24,17 +24,18 @@ function roomNumberValueAt ( tile )
...
-function powerValueAt(tile)
+function powerValueAt (tile)
+	zero = 153 -- This is middle between 0 and 256
 	if (tile == nil or tile.furniture == nil or tile.furniture.PowerConnection == nill) then
-		return 0
+		return zero
 	end
 	if (tile.furniture.PowerConnection.IsPowerConsumer) then
-		return tile.furniture.PowerConnection.InputRate * -1
+		zero = zero - tile.furniture.PowerConnection.InputRate
```
Comment (mikejbrown) [inline Assets/StreamingAssets/Overlay/overlay_functions.lua]: Maybe you want a multiplier for `tile.furniture.PowerConnection.InputRate` (and similarly below)? It's working fine but the color range just isn't that useful at the moment (powerValue only seems to vary by a range of about +/- 1 out of ~150). Or am I testing wrong?

File: Assets/Scripts/UI/Overlay/OverlayMap.cs [hunk 2]
```diff
@@ -324,6 +324,8 @@ private void Start()
         
         script = new Script();
         script.DoString(scriptTxt);
+        Type type = typeof(ModUtils);
+        script.Globals[type.Name] = type;
```
Comment (mikejbrown) [inline Assets/Scripts/UI/Overlay/OverlayMap.cs]: This whole file should probably be changed to use `LuaUtilities`, but that's for another PR. Maybe stick a TODO in here somewhere?

**Comment by Grenkin1988** [comment]
@mikejbrown Changed scailing  1 power is 12.8 for texture

**Comment by sboigelot** [comment]
@Grenkin1988  When testing this PR I'm getting this issue after opening the power overlay. I can't tell if master had the same issue as the power overlay on master is what you are trying to fix:

ScriptRuntimeException: cannot access field powerValue of userdata<Furniture>
MoonSharp.Interpreter.Execution.VM.Processor.ExecIndex (MoonSharp.Interpreter.Execution.VM.Instruction i, Int32 instructionPtr)
MoonSharp.Interpreter.Execution.VM.Processor.Processing_Loop (Int32 instructionPtr)
UnityEngine.EventSystems.EventSystem:Update()

**Comment by mikejbrown** [comment]
@sboigelot Really? Are you sure you're on the right branch? It's working fine for me. This is the issue the PR is supposed to fix.

**Comment by Grenkin1988** [comment]
@sboigelot This is fixing this issue. You probably on wrong branch

**Comment by mikejbrown** [comment]
@Grenkin1988 let's get confirmation from @sboigelot. Other than that I'm happy to merge.

**Comment by sboigelot** [comment]
@mikejbrown @Grenkin1988 ok my bad, I was on the correct branch but a restart of unity did it. For some reason it didn't recompile, or I missread something.

The PR works fine for me

**Pull request merged**